tests/integrations/raftstore/test_scale_pool.rs

```diff
@@ -198,6 +199,72 @@ fn test_decrease_pool() {
     must_get_equal(&cluster.get_engine(1), b"k2", b"v2");
 }
 
+#[test]
+fn test_increase_apply_pool_v2() {
+    use test_raftstore_v2::*;
+    let mut cluster = new_node_cluster(0, 1);
+    cluster.pd_client.disable_default_operator();
+    cluster.cfg.raft_store.apply_batch_system.pool_size = 1;
+    let _ = cluster.run_conf_change();
+    std::thread::sleep(std::time::Duration::from_millis(200));
+
+    let region = cluster.get_region(b"");
+    cluster.must_split(&region, b"k10");
+    let region = cluster.get_region(b"k11");
+    cluster.must_split(&region, b"k20");
+    let region = cluster.get_region(b"k21");
+    cluster.must_split(&region, b"k30");
+
+    fail::cfg("before_handle_tasks", "1*pause").unwrap();
+    put_with_timeout(&mut cluster, 1, b"k35", b"val", Duration::from_secs(2)).unwrap_err();
+
+    {
+        let sim = cluster.sim.rl();
+        let cfg_controller = sim.get_cfg_controller().unwrap();
+        let change = {
+            let mut change = HashMap::new();
+            change.insert("raftstore.apply-pool-size".to_owned(), "4".to_owned());
+            change
+        };
+
+        cfg_controller.update(change).unwrap();
+        std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
How to check the apply pool size is actually 4 after the update.

Reply:
It's hard to directly verify the pool size as the yatp generally creates more threads than needed.
Here I block one pool thread and make sure apply will be blocked. Then I change the pool size and after that, the following apply will not be blocked.

Reply:
How about use fail point in resize_apply_pool and let the code pass the thread count to that fail point. 
Then we can verify if that thread count is correct.